packages/react-components/src/components/RichTextEditor/Plugins/CopyPastePlugin.tsx

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { PluginEvent, EditorPlugin } from 'roosterjs-editor-types';
+import { CompatiblePluginEventType, CompatiblePasteType } from 'roosterjs-editor-types-compatible';
+
+/**
+ * CopyPastePlugin is a plugin for handling copy and paste events in the editor.
+ */
+export default class CopyPastePlugin implements EditorPlugin {
+  getName(): string {
+    return 'CopyPastePlugin';
+  }
+
+  initialize(): void {}
+
+  dispose(): void {}
+
+  onPluginEvent(event: PluginEvent): void {
+    // We don't support the pasting options such as paste as image yet.
+    if (event.eventType === CompatiblePluginEventType.BeforePaste && event.pasteType === CompatiblePasteType.Normal) {
+      event.fragment.querySelectorAll('img').forEach((image) => {
+        image.parentElement?.remove();
```

Comment:
Can there be a situation when we lose some info because of this line?
For example when we try to paste something like
```
<div>
<p>Text</p>
<img ... />
</div>
```

Reply:
When we copy from Teams, there's always a parent wrap to an image. However, this component could be used for other scenarios as well. I've updated the code to check whether the parent only has one child.

Reply:
Could you also add a commend about the reason for this line?
Also, should we check all parents nodes in this check(image node parent, parent of image node parent and so on) and delete all of them?